### ISSUE TITLE
regex_replace_chars is not inherited by context (follow-up on #55)

### ISSUE THREAD

**Issue by rfvermut** [cloudposse/terraform-null-label#57]
I believe it didn't fix the problem. You forgot `regex_replace_chars_or_context` and as result [this line](https://github.com/cloudposse/terraform-null-label/pull/56/files#diff-7a370d8342e7203b805911c92454f0f4R29) is always setting default regex_chars.

**Comment by aknysh** [comment]
I think it's ok.
```
regex_replace_chars                    = "${var.regex_replace_chars != "" ? var.regex_replace_chars : local.regex_replace_chars_context_or_default}"
```

will take it from the var if provided, or from context_or_default.

we don't need four local vars for this (as we do for name, namesoace, etc.) since we are not replacing the chars in `regex_replace_chars`

**Comment by rfvermut** [comment]
But the var is *always* provided. It has a default value.

On Monday, April 8, 2019, Andriy Knysh wrote:
> I think it's ok.
> regex_replace_chars                    = "${var.regex_replace_chars != "" ? var.regex_replace_chars : local.regex_replace_chars_context_or_default}"
> 
> will take it from the var if provided, or from context_or_default.
> we don't need four local vars for this (as we do for name, namesoace, etc.) since we are not replacing the chars in regex_replace_chars
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.￼

-- 
Sent from my Jolla

**Comment by rfvermut** [comment]
See example:
```
module "base_label" {
  source              = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.7.0"
  namespace           = "com.dotlover"
  environment         = "staging"
  regex_replace_chars = "/[^a-zA-Z0-9-.]/"
  delimiter           = "."
}

module "dns_label" {
  source  = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.7.0"
  context = "${module.base_label.context}"
  name    = "dns"
}

output "base" {
  value = "${module.base_label.name}"
}

output "dns" {
  value = "${module.dns_label.name}"
}
```

It will output `com.dotlover.staging` and `comdotlover.staging.dns` (notice the dot missing). Is this how you meant it to work?

**Comment by aknysh** [comment]
@rfvermut  thanks, you are correct.
`regex_replace_chars` is different from all other attributes (namespace, stage, name) because it always has the default value. This needs to be fixed. There are two solutions here:

1. When using `context`, the user needs to set `regex_replace_chars = ""` for the it to work (not a pretty solution for the user)

2. Since it's not possible to detect whether the `regex_replace_chars` value was used from default or was provided, we can introduce an internal local var `regex_replace_chars_default` and set it to the current default. Then in the module check if the provided value in the variable is different from the internal default; if different, use the provided value from the input; if the same, check and use the context

**Comment by Nuru** [comment]
I am closing this as I think it has been fixed by #99 if not before. In any case, it does not reflect the current design. Please open a new issue if you still have this problem.
